WordPress/openverse
# Raise exception for misclassified items in MediaStore and get filetype from Wikimedia's API

## Fixes



Related to #4717 by @obulat
Follow up #4772

## Description


Here, I broadened the number of file types the existing `commons.extensions.extract_filetype` function can guess using the `mimetypes` library. It is based on the same logic of guessing it from the URL. The `MediaStore` class will raise an error if it finds an item misclassified within the types we support.

The Wikimedia Commons DAG now gets the `filetype` from the API, giving us more accurate data in a straightforward way. You can take a look at their [MimeMap](https://doc.wikimedia.org/mediawiki-core/master/php/MimeMap_8php_source.html) file shared by @obulat below in comments. This helped complete the list of mime types and aliases.


## Testing Instructions

Check tests and confirm everything makes sense and the types that matter are covered. Optionally, run the `wikimedia_commons_workflow` DAG to verify it continues to work correctly.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
